File: tests/dynamic_fixtures/js/xss_positive.js
// XSS — positive fixture.
// Vulnerable: echoes raw user input into HTML output without escaping.
// Entry: renderPage(userInput)  Cap: HTML_ESCAPE
// Expected verdict: Confirmed (<script>NYX_XSS_CONFIRMED</script> echoed)

function renderPage(userInput) {
    process.stdout.write('__NYX_SINK_HIT__\n');
    // Unescaped output — script tags pass through verbatim.
    process.stdout.write('<html><body>' + userInput + '</body></html>\n');
}

module.exports = { renderPage };

File: tests/dynamic_fixtures/java/xss_positive.java
// XSS — positive fixture.
// Vulnerable: echoes raw user input into HTML output without escaping.
// Entry: Entry.renderPage(String)  Cap: HTML_ESCAPE
// Expected verdict: Confirmed (<script>NYX_XSS_CONFIRMED</script> echoed)

public class Entry {
    public static void renderPage(String userInput) {
        System.out.print("__NYX_SINK_HIT__\n");
        System.out.print("<html><body>" + userInput + "</body></html>\n");
    }
}
